pandas/core/indexes/period.py

```diff
@@ -674,6 +674,15 @@ def to_timestamp(self, freq=None, how='start'):
         """
         how = _validate_end_alias(how)
 
+        end = how == 'E'
+        if end:
+            if freq == 'B':
+                adjust = Timedelta(1, 'D') - Timedelta(1, 'ns')
+                return self.to_timestamp(how='start') + adjust
+            else:
```

Comment:
hmm the B looks suspect I would think that the bottom expression is correct ``(self + 1)`....

Reply:
Just speculating: is the issue with e.g. `Period('2018-03-16', freq='B').end_time` coming back as `Timestamp('2018-03-18 23:59:59.99999999')`?  That's going to need special handling, yah.

Reply:
The problem is that there are a couple of tests in ``tests/test_resample.py`` that use ``freq='B'`` that fail. For example, result and expected are different here:

```
In [2]: rng = pd.period_range('1/1/1990' , '12/31/1995', freq='W-FRI')                           
In [3]: ts = pd.Series(np.random.randn(len(rng)), index=rng) 
In [4]: result = ts.resample('B', convention='end').ffill()
In [5]: result.head()
1990-01-05    0.248627                                                     
1990-01-08    0.248627 
1990-01-09    0.248627                                                                         
1990-01-10    0.248627  
1990-01-11    0.248627            
Freq: B, dtype: float64    

In [6]: expected = result.to_timestamp('B', how='end')
In [7]: expected = expected.asfreq('B', 'ffill').to_period()
In [8]: expected.head()
Out[8]:
1990-01-08    0.248627 
1990-01-09    0.248627  
1990-01-10    0.248627                       
1990-01-11    0.248627                          
1990-01-12    0.248627                                  
Freq: B, dtype: float64   
```
But when I use:
```
if freq == 'B':
    adjust = Timedelta(1, 'D') - Timedelta(1, 'ns')
    return self.to_timestamp(how='start') + adjust
else:                                                    
    adjust = Timedelta(1, 'ns')                               
    return (self + 1).to_timestamp(how='start') - adjust   
```
it fixes the problem but I'm not sure why.

Reply:
ok, this works because we need to rollforward (the + adjust) to make sure we land on a B date.

can you add a nice comment here (about what is going on / why)

Reply:
actually see my comment below. this might work if you add an offset (a Day), which handles the freq move, rather than a Timedelta.

Reply:
Do you mean that I should try replacing ``Timedelta(1, 'D')`` with ``Day()`` to get:
```
if freq == 'B':
    adjust = Day() - Timedelta(1, 'ns')
    return self.to_timestamp(how='start') + adjust
else:                                                    
    adjust = Timedelta(1, 'ns')                               
    return (self + 1).to_timestamp(how='start') - adjust
```
If so, it doesn't seem to affect anything (all tests pass using either ``Timedelta(1, 'D')`` or ``Day()``).

Another option I considered was to try to condense the if/else in the above code into one case:
```
adjust = frequencies.to_offset(freq) - Timedelta(1, 'ns')                               
return self.to_timestamp(how='start') - adjust
```
but it didn't work because some offsets don't support subtraction of a Timedelta, I think.